classes/tests/test_events.py

```diff
@@ -8,52 +10,69 @@
     LostTrip,
     World,
 )
-import random
 from globals import ITERATION_LENGTH_MINUTES
 
 
 class EventsTests(unittest.TestCase):
     def setUp(self) -> None:
         self.world = World(40)
+        self.vehicle = self.world.state.vehicles[0]
         self.large_world = World(40, sample_size=500, number_of_clusters=20)
+        self.vehicle_large_world = self.large_world.state.vehicles[0]
         self.departure_time = 1
         self.travel_time = 5
 
     def test_scooter_departure(self):
         departure_event = ScooterDeparture(
-            self.departure_time, self.world.state.current_cluster.id
+            self.departure_time, self.vehicle.current_location.id
         )
-
+        # Remove all scooters from the scooter departure location
+        self.vehicle.current_location.scooters = []
         departure_event.perform(self.world)
 
         # test if the time of world object is set to the departure time
         self.assertEqual(departure_event.time, self.world.time)
 
-        arrival_event = next(iter(self.world.stack))
+        # Check that a lost trip event is created
+        self.assertIsInstance(self.world.stack.pop(), LostTrip)
+
+        # create new departure event with scooters in departure cluster
+        new_destination = random.choice(
+            [
+                cluster
+                for cluster in self.world.state.clusters
+                if len(cluster.scooters) > 0
+                and cluster.id != self.vehicle.current_location.id
+            ]
+        )
+        departure_event = ScooterDeparture(11, new_destination.id)
+        departure_event.perform(self.world)
+
+        arrival_event = self.world.stack.pop()
 
         # test if the arrival event created in departure has the same departure cluster id
         self.assertEqual(
-            departure_event.departure_cluster_id, arrival_event.departure_cluster_id,
+            new_destination.id, arrival_event.departure_cluster_id,
         )
 
         # scooter should have been removed from the scooters in the state
-        self.assertFalse(
-            self.world.state.get_scooters().__contains__(arrival_event.scooter)
-        )
+        self.assertNotIn(arrival_event.scooter, self.world.state.get_scooters())
 
     def test_scooter_arrival(self):
-        scooter = self.world.state.current_cluster.get_valid_scooters(20.0)[0]
+        # Take two random clusters
+        departure_cluster = self.world.state.get_random_cluster()
+        arrival_cluster = self.world.state.get_random_cluster(exclude=departure_cluster)
 
+        # Find a scooter to move in the departure cluster
+        scooter = random.choice(departure_cluster.scooters)
```

Comment:
Should be available_scooters(), since some scooters may have 0 battery percentage.